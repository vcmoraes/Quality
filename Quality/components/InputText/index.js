import React from 'react'
import styled from 'styled-components'

const TextInput = styled.TextInput`
  padding-horizontal: 7px;
  width: 100%;
  height: 40px;
  border-color: gray;
  border-width: 1px;
  border-radius: 5px;
`
class InputText extends React.Component {
  state = { text: '' }
  handlerChangeText = text => {
    switch (this.props.keyboardType) {
      case 'numeric':
      case 'decimal-pad':
      case 'number-pad':
        let value = Number(text)
        if (value <= (this.props.maxValue || Number.MAX_SAFE_INTEGER)) {
          this.setState({ text: (value && value.toString()) || '' }, () => {
            if (this.props.onChangeText) {
              this.props.onChangeText(this.state.text)
            }
          })
        }
        break
      default:
        this.setState({ text })
        if (this.props.onChangeText) {
          this.props.onChangeText(text)
        }
        break
    }
  }
  render() {
    return (
      <TextInput
        {...this.props}
        onChangeText={text => this.handlerChangeText(text)}
        value={this.state.text}
      />
    )
  }
}

export default InputText
