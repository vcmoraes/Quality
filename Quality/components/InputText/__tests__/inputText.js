import React from 'react'
import InputText from '../index'
import renderer from 'react-test-renderer'

describe('Components - InputText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<InputText />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
