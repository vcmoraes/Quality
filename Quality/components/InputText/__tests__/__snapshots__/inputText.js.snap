// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - InputText renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "paddingHorizontal": 7,
        "width": "100%",
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
